lib/rules/declaration-block-no-redundant-longhand-properties/index.js

```diff
@@ -22,8 +22,49 @@ const meta = {
 	fixable: true,
 };
 
+const customResolvers = new Map([
+	[
+		'grid-template',
+		(/** @type {Map<string, Declaration>} */ transformedDeclarationNodes) => {
+			const areas = transformedDeclarationNodes.get('grid-template-areas')?.value.trim();
+			const columns = transformedDeclarationNodes.get('grid-template-columns')?.value.trim();
+			const rows = transformedDeclarationNodes.get('grid-template-rows')?.value.trim();
+
+			if (areas === undefined || columns === undefined || rows === undefined) return;
+
+			const splitAreas = [...areas.matchAll(/".+?"/g)].map((x) => x[0]);
```

Comment:
[suggest] How about changing to a bit more accurate regex?

```suggestion
			const splitAreas = [...areas.matchAll(/"[^"]+?"/g)].map((x) => x[0]);
```